x/merkledb/path.go

```diff
@@ -306,6 +271,55 @@ func (p Path) Skip(tokensToSkip int) Path {
 	return result
 }
 
+func (p Path) AppendExtend(token byte, path Path) Path {
+	appendBytes := p.bytesNeeded(p.tokensLength + 1)
+	totalLength := p.tokensLength + 1 + path.tokensLength
+	buffer := make([]byte, p.bytesNeeded(totalLength))
+	p.appendIntoBuffer(buffer[:appendBytes], token)
+
+	// the extension path will be shifted based on the number of tokens in the partial byte
+	tokenRemainder := (p.tokensLength + 1) % p.tokensPerByte
+	path.extendIntoBuffer(tokenRemainder, buffer[appendBytes-1:])
```

Comment:
I think it might be cleaner to just have the logic of `extendIntoBuffer` here and remove that method. I find it a bit confusing that we call a method on `p` and then on `path`. That method isn't used elsewhere.